### PR TITLE
All internal components are renamed to Develocity

### LINKED ISSUE #584: When linking to documentation the short link is not always used
The short link `https://gradle.com/bvs` redirects to `https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/`. We should always use the short form where possible.

Here are all the places where it is referenced:

https://github.com/search?q=repo%3Agradle%2Fgradle-enterprise-build-validation-scripts+%22https%3A%2F%2Fgithub.com%2Fgradle%2Fgradle-enterprise-build-validation-scripts%2Fblob%22&type=code

### PR DESCRIPTION
**Pull request by erichaagdev**
This PR updates all internal references from Gradle Enterprise to Develocity.

Much of this is just a simple rename of `ge` and `gradle enterprise` to `develocity`, but there were a few occurrences where this wasn't the case. I have outlined any noteworthy and exceptional cases below:

- The bundled Maven extension was renamed from `configure-gradle-enterprise-maven-extension` to `configure-develocity-maven-extension` (it still configures both)
- All system properties passed to `configure-develocity-maven-extension` had their prefixes updated from `com.gradle.enterprise.build-validation` to `develocity.build-validation`
  - For Gradle, this has already been done, setting aside the fact that Gradle now configures the init script using environment variables
  - I made certain to **not rename** the system properties that are configuring the "Gradle Enterprise" Maven extension directly, such as `gradle.enterprise.url` and `gradle.enterprise.allowUntrustedServer` as these must remain for as long as we support using the GE Maven extension with the scripts
- All links pointing to documentation in `Gradle.md` or `Maven.md` has been updated from `https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob` to the special short form redirect `https://gradle.com/bvs`
  - While this was not strictly necessary, it saves us one renaming step in the future and conveniently resolves #584

What remains are things that cannot yet be updated. This primarily includes all references to the repository, Gradle Enterprise distributions, bundled GE extension.

### CODE REVIEW

**Review by ribafish — APPROVED** [review]
LGTM 👍

File: build.gradle.kts [hunk 1]
```diff
@@ -87,7 +87,7 @@ dependencies {
     develocityInjection("com.gradle:develocity-injection:1.0")
     develocityComponents("com.gradle:build-scan-summary:$buildScanSummaryVersion")
     develocityMavenComponents("com.gradle:gradle-enterprise-maven-extension:1.18.4")
```
Comment (jprinet) [inline build.gradle.kts:89]: Is this gradle-enterprise artifact expected there?

**Review by jprinet — COMMENTED** [review]
Great job, just a question about a GE artifact

**Review by etiennestuder — APPROVED** [review]
LGTM. Thanks, Eric.

**Pull request merged**